gips/0083.md

```diff
@@ -0,0 +1,240 @@
+
+Future GIPs will be proposed to enable more complex pricing strategies, such as pay-per-cpu-time, pay-per-storage, pay-per-query, pay-per-block, pay-per-attestation, etc.
+
+### Curation
+
+THIS SECTION IS NON-NORMATIVE
+
+Curation can happen on the published Subgraph manifest above, although a nonce may be added to avoid paying curation fees. Because there are no indexer rewards, curation is not adding value, therefore it is reasonable for participants to want to avoid it.
+
+
+
+## Economic Security and Disputes
+
+Economic security is achieved through slashing and disputes, similar to subgraphs as of Jan 24th 2025. Indexers providing incorrect data CAN have their stake slashed. Therefore, to be an Indexer recognized by the Payment Gateway, one must staked the Network token, just like for subgraphs.
+
+The means are similar to subgraphs, by providing signed attestations, attached to allocations, and having an Arbiter validate those attestations, and slash the Indexer's stake if necessary.
+
+### Attestations
+
+Having no queries like subgraphs have, Proofs of Indexing for Substreams are always null.
+
+> [!NOTE]
+> This is why enabling indexing rewards for Substreams is not part of this GIP. Future GIPs, potentially with Horizon, will propose a mechanism to reward Indexers for providing Substreams services.
+
+Substreams endpoints provide signed attestations with each block's response, as per [the `attestation` field](https://github.com/streamingfast/substreams/blob/develop/proto/sf/substreams/rpc/v2/service.proto#L96) in the `BlockScopedData` response, matching the operator key specified in [the `attestation_public_key` field](https://github.com/streamingfast/substreams/blob/develop/proto/sf/substreams/rpc/v2/service.proto#L106) of the `SessionInit` response.
+
+The payload signed over is composed of:
+- the Substreams output module's payload, hashed with SHA256 (32 bytes), followed by:
+- the 'b' character, followed by:
+- the block ID as UTF-8 encoded string, followed by:
+- the 'm' character, followed by:
+- the module hash for the top-level Substreams module being attested (32 bytes)
+
+The attestation payload MUST be signed by a special _attestation key_, derived from the operator's key, so that disputes can be opened on the _Substreams Service Deployment ID_.
+
+A valid _attestation key_, attached to such an allocation MUST be verified by the Payment Gateway, before giving the assurance to Consumers that the Indexer can be slashed for misbehaving.
+
+An Arbiter MUST be able to validate payloads, through multiple providers and investigation, and after judgement, be able to slash the Indexer's stake, using the information provided in those payloads via an on-chain allocations.
+
+> INFORMATIVE: The module hash covers everything needed for deterministic execution, and nothing more. It is easily computed by all known compliant Substreams library.
+
+
+#### Methods of analysis
+
+THIS SECTION IS NON-NORMATIVE
```

Comment:
I think it would be nice to add some basic description of how signed attestations is submitted and general timeline for arbitration